self-core-impl/src/main/java/com/selfxdsd/core/GitlabIssue.java

```diff
@@ -0,0 +1,214 @@
+import com.selfxdsd.api.storage.Storage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.json.JsonObject;
+import javax.json.JsonValue;
+import java.net.URI;
+
+/**
+ * An Issue in a Gitlab repository.
+ * <br/>
+ * <a href="https://docs.gitlab.com/ee/api/issues.html#single-project-issue">
+ *  See specification </a>
+ *  <br/>
+ *  For a merge request
+ *  <a href ="https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr">
+ *     See specification </a>
+ * @author criske
+ * @version $Id$
+ * @since 0.0.38
+ * @todo 719#60 Implement and test method `assign()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 720#60 Implement and test method `unassign()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 721#60 Implement and test method `close()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 722#60 Implement and test method `reopen()` for
+ *  GitlabIssue by following GithubIssue as model.
+ * @todo 723#60 Start implementing GitlabComments by following GithubComments
+ *  as model. These will be used by GitlabIssue.
+ * @todo 724#60 Start implementing GitlabLabels by following GithubLabels
+ *  as model. These will be used by GitlabIssue.
+ */
+final class GitlabIssue implements Issue {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        GitlabIssue.class
+    );
+
+    /**
+     * Issue base uri.
+     */
+    private final URI issueUri;
+
+    /**
+     * Issue JSON as returned by Gitlab's API.
+     */
+    private final JsonObject json;
+
+    /**
+     * Self storage, in case we want to store something.
+     */
+    private final Storage storage;
+
+    /**
+     * Gitlab's JSON Resources.
+     */
+    private final JsonResources resources;
+
+    /**
+     * Ctor.
+     * @param issueUri Issues base URI.
+     * @param json Json Issue as returned by Gitlab's API.
+     * @param storage Storage.
+     * @param resources Gitlab's JSON Resources.
+     */
+    GitlabIssue(
+        final URI issueUri,
+        final JsonObject json,
+        final Storage storage,
+        final JsonResources resources
+    ) {
+        this.issueUri = issueUri;
+        this.json = json;
+        this.storage = storage;
+        this.resources = resources;
+    }
+
+    @Override
+    public String issueId() {
+        return String.valueOf(this.json.getInt("iid"));
+    }
+
+    @Override
+    public String provider() {
+        return Provider.Names.GITLAB;
+    }
+
+    @Override
+    public String role() {
+        final String role;
+        if(this.isPullRequest()) {
+            role = Contract.Roles.REV;
+        } else {
+            role = Contract.Roles.DEV;
+        }
+        return role;
+    }
+
+    @Override
+    public String repoFullName() {
+        return this.json
+            .getJsonObject("references")
+            .getString("full")
+            .split("[#!]")[0];
```

Comment:
@amihaiemil we're using a regexed split cause
Issue json is using `#`
```json
"references": {
    "short": "#1",
    "relative": "#1",
    "full": "my-group/my-project#1"
  }
``` 
Pull request (merge request) is using `!`
```json
"references": {
    "short": "!1",
    "relative": "!1",
    "full": "my-group/my-project!1"
  }
```